services/student-group.service.js

```diff
@@ -0,0 +1,59 @@
+import { saveGroup, getAllGroups, getGroup, updateGroup, deleteGroup } from "../repository/index.js";
+import AppError from "../utils/appError.js";
+
+export const saveStudentGroup = async (data) => {
+  let { groupID, groupName, topic, group_members, panel_members, supervisor, co_supervisor, supervisor_status, co_supervisor_status } = data;
+  try {
+    const group = await saveGroup({ groupID, groupName, topic, group_members, panel_members, supervisor, co_supervisor, supervisor_status, co_supervisor_status });
```

Comment:
Put the data object here without curly braces. Do the same for UpdateStudentGroup.